packages/devtools_app/lib/src/network/http_request_data.dart

```diff
@@ -0,0 +1,246 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'http.dart';
+
+/// Contains all state relevant to completed and in-progress HTTP requests.
+class HttpRequests {
+  HttpRequests({this.requests, this.outstanding}) {
```

Comment:
why not just use default values for the parameters?

Reply:
+1 to default combined with an assert that the parameter is not null.